packages/react/src/components/Tabs/Tabs.js

```diff
@@ -139,35 +146,47 @@ export default class Tabs extends React.Component {
     };
   };
 
+  getDirection = evt => {
+    if (match(evt, keys.ArrowLeft)) {
+      return -1;
+    }
+    if (match(evt, keys.ArrowRight)) {
+      return 1;
+    }
+    return 0;
+  };
+
+  getNextIndex = (index, direction) => {
+    const enabledTabs = this.getEnabledTabs();
```

Comment:
Would we want to use this helper at all: https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/internal/keyboard/navigation.js#L27

Reply:
that doesn't look like a drop-in replacement for the function I have currently

Reply:
Just was speaking to figuring out the `nextIndex` and using a ring buffer instead of the current logic 👍

Reply:
how does it account for disabled tab indices?

Reply:
@emyarod if I understand correctly, the ring buffer would be done on a filtered down version of tabs, correct?

```js
const tabs = [
  {
    title: 'Tab A',
    disabled: false,
  },
  {
    title: 'Tab B',
    disabled: true,
  },
  {
    title: 'Tab B',
    disabled: false,
  },
];

const enabledTabs = tabs.filter(tab => !tab.disabled);
getNextIndex(ArrowRight, 0, enabledTabs.length); // 1
getNextIndex(ArrowRight, 1, enabledTabs.length); // 0
getNextIndex(ArrowLeft, 0, enabledTabs.length); // 1
```

Definitely get that it's not mapping exactly 1:1, but just was seeing if the pattern was worthwhile to match 👍

Reply:
the returned index from that function corresponds to the element's location in the filtered array and not the full array. but if you are just referring to the pattern then I have implemented it already, I'm just also accounting for the fact that the component is expecting the full array and not only the enabled tabs